### ISSUE TITLE
How to extract "MONEY" using de_core_news_sm

### ISSUE THREAD

**Issue by leugh** [explosion/spaCy#2826]
As spaCy's "de_core_news_sm" doesn't support MONEY entity I am wondering if somebody has tried doing the same and could share some advice.

Specifically, if using the POS-tagger and searching for patterns is enough or if it would make sense to train the de_core_news_sm model to recognize MONEY entities?

I am very new to ML in general so I am hoping that somebody could give me some advice - despite the very broad question.

## Your Environment
* **spaCy version:** 2.0.12
* **Platform:** Darwin-16.7.0-x86_64-i386-64bit
* **Python version:** 3.6.5
* **Models:** de

**Comment by ines** [comment]
Hi! The annotation scheme used in the models always depends on the data that's available. For the English model, we were able to train from a corpus that comes with a much wider range of entity types. For the other languages, we only had Wikipedia data available that was annotated with the much more limited label scheme.

Going forward, we want to create our own data resources, so we can use the more extensive entity types across more languages. (Basically, we're planning to run our own annotation projects using our tool [Prodigy](https://prodi.gy) and re-label the training corpora used for the non-English models – e.g. the TIGER corpus for German –, but with the more complex entity label scheme. But this is obviously going to be a larger and more involved project.)

> Specifically, if using the POS-tagger and searching for patterns is enough or if it would make sense to train the de_core_news_sm model to recognize MONEY entities?

Writing patterns for spaCy's [rule-based `Matcher`](https://spacy.io/usage/linguistic-features#rule-based-matching) is definitely something you should try first – because even if you decide to train a model later on, you can use those patterns to bootstrap training data, which will be incredibly valuable. I'd also recommend creating an evaluation set you can use to check how well your patterns are working – and how your model is comparing, if you decide to train one.

For `MONEY`, token-based patterns could potentially work quite well, because many of the mentions you're looking for follow a fairly consistent pattern (e.g. a `is_digit` / `like_num` token followed by a currency symbol or word).

[This docs section](https://spacy.io/usage/training#training-data) has some background on how to get training data, and how to use match patterns to bootstrap training sets. Once you have a good number of examples in context (a few hundred to a few thousand), you can use those to update the existing `de_core_news_sm` model. 

One thing that's important to keep in mind, though: you don't want the model to overfit on the new data and "forget" all the other entity types it previously recognised. So you always want to include examples of the previous entities that were recognised correctly. You can do this by running the existing model over some text, keeping the correct entity predictions, adding your new `MONEY` annotations and then updating the model with the new data. And always make sure to hold data back for evaluation, so you can get a good picture for whether your model is improving or not!
